gbtis/Kinect.cs

```diff
@@ -127,6 +141,26 @@ public class Kinect {
         }
 
         /// <summary>
+        /// Calculates the average finger position over the last
+        /// POINT_SAMPLE_SIZE frames
+        /// </summary>
+        /// <param name="x">The latest x coordinate</param>
+        /// <param name="y">The latest y coordinate</param>
+        /// <returns>The average position</returns>
+        private Point getAverageFingerTipPosition(float x, float y) {

```

Comment:
Compared to all the other stuff we're doing at 30fps, probably not?  Although I will admit that it is not the most efficient solution and I do like your suggestion.  I think the best option is to modify your algorithm slightly:

```C#
rollingAverage(double new_sample) {
    avg -= samples.RemoveFirst() / N;
    avg += new_sample / N;
    samples.addLast(new_sample);

    return avg;
}
```
Since your algorithm doesn't account account for the order in which data points come in it doesn't give a true rolling average (Yours gives the average over all points, when we want to forget about them after so long).  In this solution I'd still need the two lists of previous x, y points but since removeFirst() and addLast() are both O(1) it would be way more efficient than my current implementation.  

Thoughts?